### PR TITLE
Ireading for teachers

### PR DESCRIPTION
**Pull request by jivewise**
screen shot 2015-02-26 at 11 46 09 pm
screen shot 2015-02-26 at 11 45 58 pm
screen shot 2015-02-26 at 11 45 47 pm

Adding in initial component for creating iReadings.

### CODE REVIEW

File: src/components/teacher/reading.cjsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    @setState({
+      isModalOpen: !@state.isModalOpen
+    })
+
+  render: ->
+    <BS.Button onClick={@handleToggle} bsStyle="primary">Edit Readings</BS.Button>
+
+  renderOverlay: ->
+    if !@state.isModalOpen
+      return <span/>
```
Comment (philschatz) [inline src/components/teacher/reading.cjsx]: Would an empty return still work?

File: src/components/teacher/reading.cjsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    dueDate = if @state.due_at then @state.due_at else @props.due_at
+
+    <BS.Panel bsStyle="default" className="create-reading" footer={footer}>
+      <h1>{headerText}</h1>
+      <div>
+        <label for="title">Name</label>
+        <input id="title" type="text" value={@props.title}/>
+      </div>
+      <div>
+        <label for="due-date">Due Date</label>
```
Comment (philschatz) [inline src/components/teacher/reading.cjsx]: Should this be `htmlFor="..."`?

File: src/router.coffee [hunk 3]
```diff
@@ -10,6 +11,8 @@ routes = (
     <Route path='dashboard' name='dashboard' handler={Dashboard} />
     <Route path='tasks' name='tasks' handler={Tasks} />
     <Route path='tasks/:id' name='task' handler={SingleTask} />
+    <Route path='reading' name='createReading' handler={CreateReading} />
```
Comment (philschatz) [inline src/router.coffee]: This should probably be plural to match `/tasks/`

Comment (jivewise) [inline src/router.coffee]: I updated it to be /readings, but still like the non-plural form better :).

I also noticed something weird with routes; if I go to /readings/ or /tasks/, I get the invalid state.  Is there a way we can avoid having to add duplicate routes for those?

File: src/components/teacher/reading.cjsx (hunk 1 above)
Comment (jivewise) [inline src/components/teacher/reading.cjsx]: No, looks like we need to return something there in order for this to work.  Just the way the react-bootstrap modal component works.

**Pull request merged**